zerver/tests/test_subs.py

```diff
@@ -3516,7 +3516,7 @@ def test_compose_views_rollback(self) -> None:
         def thunk1() -> HttpResponse:
             user_profile.full_name = "Should not be committed"
             user_profile.save()
-            return json_success()
+            return json_success(request)
```

Comment:
I think I handled this wrong, resulting in CI failures in main -- I think we wanted to remove the `json_success` from this function altogether when changing the `compose_views` API.

Reply:
@timabbott Yeah, the `thunks` being called in `compose_views` are view functions that return `json_success` when everything works (and that rollback the database for any changes made / raise `JsonableError` when there's an issue). So, I'm a bit stuck on how we could eliminate the `json_success` from those single add / delete views.

We could put the rollback and `thunks` loop back into the two view functions using `compose_views`. It looks like @showell  refactored this to be a separate function in `streams.py` about 6 years ago and it was added to `user_groups.py` about a year after that. If we made that change then we'd need to test the rollback within the two views vs testing the helper. I guess can see doing that if we feel like these types of endpoints won't be common / increasing, but there might also be considerations I'm not aware of as well.

In the use case for `streams.py`, the `thunk` views are also actively used in other rest paths in `zproject.urls.py`. Also, that [update subscriptions endpoint using `compose_views`](https://zulip.com/api/update-subscriptions) is documented in the OpenAPI, but is not in the api docs sidebar so not easily find-able for folks looking at the documentation site. So maybe it's not an endpoint we're encouraging clients to use? And eventually eliminating?

In the use case for `user_groups.py`, the `thunk` views are currently only used for the [one rest path](https://zulip.com/api/update-user-group). That endpoint is documented and in the sidebar list in this case. This seems like an endpoint that is encouraged for use, but also would be easier to refactor since the add / delete view functions aren't being used outside of the one endpoint.

Does this merit a CZO topic conversation? I can move this all there if that makes the most sense. Either to backend or api_design ...

Reply:
I think #backend is probably fine for this, since it's an implementation detail.  Seems worth raising there; comments on closed GitHub issues tend to get lost :).

But yeah, it's very possible we want to deprecate endpoints using this pattern; it's a bit of a weird one dating to very early in Zulip's history.